infra/build/functions/fuzzbench.py

```diff
@@ -0,0 +1,154 @@
+    return []
+
+  config = build_project.Config(config.testing, None, config.repo,
+                                config.branch, config.parallel, config.upload)
+
+  # TODO(metzman): Make this a command line argument
+  fuzzing_engine = 'libafl'
+
+  steps = [
+      {
+          'args': [
+              'clone', 'https://github.com/google/fuzzbench', '--depth', '1',
+              FUZZBENCH_PATH
+          ],
+          'name': 'gcr.io/cloud-builders/git',
+          'volumes': [{
+              'name': 'fuzzbench_path',
+              'path': FUZZBENCH_PATH,
+          }],
+      },
+      {
+          'name': 'gcr.io/cloud-builders/docker',
+          'args': ['pull', 'gcr.io/oss-fuzz-base/base-builder-fuzzbench']
+      },
+      {  # TODO(metzman): Don't overwrite base-builder
+          'name':
+              'gcr.io/cloud-builders/docker',
+          'args': [
+              'tag', 'gcr.io/oss-fuzz-base/base-builder-fuzzbench',
+              'gcr.io/oss-fuzz-base/base-builder'
+          ]
+      },
+  ]
+
+  steps += build_lib.get_project_image_steps(project.name,
+                                             project.image,
+                                             project.fuzzing_language,
+                                             config=config)
+
+  engine_dockerfile_path = os.path.join(FUZZBENCH_PATH, 'fuzzers',
+                                        fuzzing_engine, 'builder.Dockerfile')
+  build_args = [
+      'build', '--build-arg', f'parent_image=gcr.io/oss-fuzz/{project.name}',
+      '--tag', project.image, '--file', engine_dockerfile_path,
+      os.path.join(FUZZBENCH_PATH, 'fuzzers')
+  ]
+  engine_step = [
+      {
+          'name': 'gcr.io/cloud-builders/docker',
+          'args': build_args,
+          'volumes': [{
+              'name': 'fuzzbench_path',
+              'path': FUZZBENCH_PATH,
+          }],
+      },
+  ]
+  steps += engine_step
+  build = build_project.Build(fuzzing_engine, 'address', 'x86_64')
+  env = build_project.get_env(project.fuzzing_language, build)
+  env.append(f'FUZZBENCH_PATH={FUZZBENCH_PATH}')
+  env.append(f'PROJECT={project.name}')
+  env.append('OSS_FUZZ_ON_DEMAND=1')
+  compile_project_step = {
+      'name':
+          project.image,
+      'env':
+          env,
+      'volumes': [{
+          'name': 'fuzzbench_path',
+          'path': FUZZBENCH_PATH,
+      }],
+      'args': [
+          'bash',
+          '-c',
+          # Remove /out to make sure there are non instrumented binaries.
+          # `cd /src && cd {workdir}` (where {workdir} is parsed from the
+          # Dockerfile). Container Builder overrides our workdir so we need
+          # to add this step to set it back.
+          (f'ls /fuzzbench && rm -r /out && cd /src && cd {project.workdir} && '
+           f'mkdir -p {build.out} && compile'),
+      ],
+  }
+  steps.append(compile_project_step)
+  env.extend(['FUZZ_TARGET=iccprofile_atf', f'BENCHMARK={project.name}'])
```

Comment:
It would be nice to avoid having these hardcodes completely since I think it's pretty easy to pipe through arguments from CLI for this.

Reply:
It's not trivial right now because we share code with another script that does commandline parsing.